File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {

        int left = 0;
        int right = numbers.length - 1;

        while (left < right) {
            // keep track of sum
            int sum = numbers[left] + numbers[right];
            if (sum == target){
                // 1-indexed means starting from 1 not 0
                return new int[]{left + 1, right + 1};
                // if sum is too small means that left is too small
            } else if (sum < target ) {
                left++;
            } else {
                right--;
            }
        }
        return new int[0];
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        while l <= r:
            middle = (l + r) // 2 # get the current middle element 
            # if target is greater than number at middle, that means target is hiding in the numbers too the right
            # so we increment our left pointer to check past that half
            if target > nums[middle]:
                l = middle + 1
            elif target < nums[middle]:
                r = middle - 1
            else: # else nums[middle] == target so we return the index
                return middle
        return -1